plugin/trino-hive/src/test/java/io/trino/plugin/hive/coercions/TestVarcharToDoubleCoercer.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.coercions;
+
+import io.trino.spi.block.Block;
+import io.trino.spi.type.Type;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static io.airlift.slice.Slices.utf8Slice;
+import static io.trino.plugin.hive.HiveTimestampPrecision.DEFAULT_PRECISION;
+import static io.trino.plugin.hive.HiveType.toHiveType;
+import static io.trino.plugin.hive.coercions.CoercionUtils.createCoercer;
+import static io.trino.spi.predicate.Utils.blockToNativeValue;
+import static io.trino.spi.predicate.Utils.nativeValueToBlock;
+import static io.trino.spi.type.DoubleType.DOUBLE;
+import static io.trino.spi.type.VarcharType.createUnboundedVarcharType;
+import static io.trino.type.InternalTypeManager.TESTING_TYPE_MANAGER;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestVarcharToDoubleCoercer
+{
+    @Test(dataProvider = "varcharValues")
+    public void testDoubleToVarcharCoercions(String varcharValue, boolean treatNaNAsNull, Double expectedDoubleValue)
+    {
+        assertCoercions(createUnboundedVarcharType(), utf8Slice(varcharValue), DOUBLE, expectedDoubleValue, treatNaNAsNull);
+    }
+
+    @DataProvider
+    public Object[][] varcharValues()
+    {
+        return new Object[][] {
+                {"-Infinity", true, Double.NEGATIVE_INFINITY},
+                {"NaN", true, null},
+                {"NaN", false, Double.NaN},
+                {"1.12e+3", false, Double.parseDouble("1120.0")},
+                {"123456789.12345678", true, Double.parseDouble("123456789.12345678")},
+                {"123", true, Double.parseDouble("123")},
+                {"Infinity", false, Double.POSITIVE_INFINITY},
+                // above infinity
```

Comment:
Below infinity ?

Reply:
Is it possible to add them in PTs as we might need to know how it would coerce across multiple format